### PR TITLE
Add Azure SignalR binding

### PR DESCRIPTION
**Pull request by fbeltrao**
# Description

Add support to Azure SignalR as output binding

## Issue reference

#147 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Comment by yaron2** [comment]
cc @davidfowl

File: bindings/azure/signalr/signalr.go [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/dapr/components-contrib/bindings"
+	"github.com/gbrlsnchs/jwt/v3"
```
Comment (davidfowl) [inline bindings/azure/signalr/signalr.go]: Is this the defactor JWT package for golang projects?

Comment (yaron2) [reply bindings/azure/signalr/signalr.go]: If there'a a "defacto" package for JWT for Go I'd say its [this](https://github.com/dgrijalva/jwt-go) one..

@fbeltrao would it be possible to make the switch unless there are special reasons to use the current one?

Comment (fbeltrao) [reply bindings/azure/signalr/signalr.go]: Wasn’t sure about that one too. Thanks for the tip.

File: bindings/azure/signalr/signalr.go [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+	version    string
+	hub        string
+	tokens     map[string]signalrCachedToken
+	httpClient *http.Client
+}
+
+// Init is responsible for initializing the SignalR output based on the metadata
+func (s *SignalR) Init(metadata bindings.Metadata) error {
+	connectionString, ok := metadata.Properties[connectionStringKey]
+	if !ok || connectionString == "" {
```
Comment (davidfowl) [inline bindings/azure/signalr/signalr.go]: for my edification, `connectionString == ""` or `len(connectionString) == 0`? I see both patterns in this file.

File: bindings/azure/signalr/signalr.go [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+	}
+
+	// Expected: Endpoint=https://<servicename>.service.signalr.net;AccessKey=<access key>;Version=1.0;
+	connectionValues := strings.Split(strings.TrimSpace(connectionString), ";")
+	for _, connectionValue := range connectionValues {
+		if i := strings.Index(connectionValue, "="); i != -1 {
+			k := connectionValue[0:i]
+			switch k {
+			case "Endpoint":
+				s.endpoint = connectionValue[i+1:]
```
Comment (jjcollinge) [inline bindings/azure/signalr/signalr.go]: Is it worth checking i+1 exists before indexing into the array? i.e. `Endpoint=;`

File: bindings/azure/signalr/signalr.go (hunk 2 above)
Comment (fbeltrao) [reply bindings/azure/signalr/signalr.go]: According to what I [read](https://stackoverflow.com/questions/18594330/what-is-the-best-way-to-test-for-an-empty-string-in-go) the compiler will do make both [equal](https://stackoverflow.com/questions/18594330/what-is-the-best-way-to-test-for-an-empty-string-in-go/54804858#54804858), so we should use the one that is more readable.

Will change to == as I find more readable

File: bindings/azure/signalr/signalr.go (hunk 3 above)
Comment (fbeltrao) [reply bindings/azure/signalr/signalr.go]: Good catch, thanks @jjcollinge

File: bindings/azure/signalr/signalr.go [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+	return url, nil
+}
+
+func (s *SignalR) sendMessageToSignalR(url string, token string, data []byte) error {
+	httpReq, err := http.NewRequest("POST", url, bytes.NewBuffer(data))
+	if err != nil {
+		return err
+	}
+
+	httpReq.Header.Add("Authorization", "Bearer "+token)
```
Comment (davidfowl) [inline bindings/azure/signalr/signalr.go]: Set vs Add?

Comment (yaron2) [reply bindings/azure/signalr/signalr.go]: In this case, it doesn't matter.

Here are the differences:

```
Set sets the header entries associated with key to the single element value. It replaces any existing values associated with key. The key is case insensitive; it is canonicalized by textproto.CanonicalMIMEHeaderKey. To use non-canonical keys, assign to the map directly.
```

```
Add adds the key, value pair to the header. It appends to any existing values associated with key. The key is case insensitive; it is canonicalized by CanonicalHeaderKey.
```

Comment (davidfowl) [reply bindings/azure/signalr/signalr.go]: Let’s call set for both

Comment (yaron2) [reply bindings/azure/signalr/signalr.go]: I missed the Set call. Should have consistency, so either set or add for both.

**Pull request merged**